subsys/usb/Kconfig

```diff
@@ -69,6 +69,7 @@ config USB_REQUEST_BUFFER_SIZE
 	range 256 65536 if USB_DEVICE_NETWORK_RNDIS
 	range 8 65536
 	default 256 if USB_DEVICE_NETWORK_RNDIS
+	default 512 if (BT_CTLR_ADV_EXT && USB_DEVICE_BLUETOOTH)
```

Comment:
Instead of `BT_CTLR_ADV_EXT`, could we maybe do it based on `BT_RX_BUF_LEN > ?128?`
It might be that the hci_usb sample is configured with data length extension instead, providing 251 ACL data.

Reply:
I'm not sure about the size thus I count on You if 512 is enough and when to enable extended size. Maybe:
``` diff
+ 	default BT_RX_BUF_LEN if USB_DEVICE_BLUETOOTH
```
?

Reply:
Hmm, I'm not sure it would work, because the buffer might be larger than that and we can't do additions in Kconfig (atleast that I know of).
The addition is 1 byte, which is the HCI type byte.

I think maybe best would be `default 265 if BT_RX_BUF_LEN > 127`
Using 127 because 128 is the default in this selection, and 264 is the maximum BT_RX_BUF_LEN is set to based on defaults.

@jhedberg Opinions on this?

Reply:
Would the `BT_HCI_RESERVE` Kconfig option help here at all? It exists to reserve space for HCI transport headers in the beginning of the buffer? That said, I get the impression that the desire here is to extend the tail-end of the buffer to round it up to some value that's required by USB?

Reply:
Actually, the `BT_HCI_RESERVE` is intended for the host-based setups, and it seems this is about controller-only. I think the analogue for it there is `BT_HCI_RAW_RESERVE`

Reply:
Just ignore the above comments: I misunderstood this being about buffers managed by the Bluetooth stack. The suggestion with Kconfig conditionals sounds alright to me.

Reply:
I've checked now that the actual buffer is larger based on [other](https://github.com/zephyrproject-rtos/zephyr/blob/master/include/bluetooth/buf.h#L46-L55) Kconfig options. For now it seems that additon of '1' seems sufficient. @joerchan How likely is it to change?

For now I will update the PR to
default **266** if BT_RX_BUF_LEN > 127.

266 because 265 might introduce a confusion to users that this might be mistake 256-265. We are rich enough to dedicate that one additional Byte.

Reply:
> How likely is it to change?

I would say very unlikely, it is used for the 1-byte HCI identifier.